libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -1362,106 +1365,571 @@ def _extract_images_from_page(self, page: pypdfium2._helpers.page.PdfPage) -> st
                 continue
             numpy.save(image_bytes, image.get_bitmap().to_numpy())
             blob = Blob.from_data(image_bytes.getvalue(), mime_type="application/x-npy")
-            text_from_image = next(self.images_parser.lazy_parse(blob)).page_content
+            text_from_image = next(
+                self.images_parser.lazy_parse(blob)  # type: ignore
+            ).page_content
             str_images.append(
                 _format_inner_image(blob, text_from_image, self.images_inner_format)
             )
             image.close()
         return _FORMAT_IMAGE_STR.format(image_text=_JOIN_IMAGES.join(str_images))
 
 
+# The legacy PDFPlumberParser use key with upper case.
+# This is not aligned with the new convention, which requires the key to be in
+# lower case.
+class _PDFPlumberParserMetadata(dict):
```

Comment:
This feels a bit heavy handed as a solution?

Could we add an init parameter to `PDFPlumberParser` called "metadata_format" and have it act as a version? It could allow a "legacy" and "standard" versions?

Reply:
The default strategy will be "legacy" or "standard"?

Reply:
@eyurtsev 
Another solution can be to set upper and lower case keys.

Reply:
Maybe default strategy would be `None` which will resolve to `legacy` to avoid breaking user code.

This would allow emitting a warning users to alert them that they should specify a format either `legacy` or `standard`.

Having both upper and lower case keys doesn't feel like an API we'd want to end up with long-term. Having both keys in the metadata will be confusing (i.e., are both keys the same are they different why are there two etc.)

An alternative is also to not modify the metadata at all and handle this on the client side

Reply:
I implement a replication of keys. The legacy keys and lower keys. It's compatible without breaking user code.

Reply:
It's in the last version now.